src/pages/Lobby.js

```diff
@@ -22,178 +18,299 @@ import SaveAlt from '@material-ui/icons/SaveAlt';
+      next: (gameData) => {
+        let game = gameData.value.data.onUpdateGame
+        let currentGame = localStorage.getItem(CURRENT_GAME)
+        if(currentGame && currentGame === game.id){
+          this.props.history.push({pathname: '/game', state: {message: game}})
         }
-        const createGameButtonStyle = {
-            width: "30%",
-            padding: "10px", 
-            marginBottom: "10px",
-            backgroundColor: '#333333',
-            color: '#FFF',
-            fontFamily: "AppleSDGothicNeo-Bold"
+        else if(this.gamesData.hasOwnProperty(game.id)){
+          delete this.gamesData[game.id]
+          this.removeGameFromLobby(game.id)
         }
-        return (
-            <Container maxWidth='sm' style={lobbyStyle}>
-                {/* <Button style={createGameButtonStyle} variant="contained" onClick={this.props.makeDialogVisible}>
-                    Create a game
-                </Button> */}
-                <CreateGameDialog 
-                    onClick={this.props.makeDialogVisible}
-                 />
-                <div style={{width: "100%"}}>
-                <MaterialTable
-                    icons = {tableIcons}
-                    columns={lobbyColumns}
-                    data={this.state.games}
-                    title='Lobby'
-                    maxWidth="md"
-                    options={{
-                        headerStyle: {
-                            backgroundColor: '#FFF',
-                            fontFamily: "AppleSDGothicNeo-SemiBold, verdana",
-                            fontSize: "18px",
-                            color: '#333333'
-                        },
-                        paging: false,
-                        searchFieldStyle:{
-                            fontSize: "14px",
-                            fontFamily: 'verdana'
-                        }
-                    }}
-                    localization = {{
-                        toolbar: {
-                            searchPlaceholder: "keywords"
-                        }
-                    }}
-                />
-                </div>
-            </Container>
-        )
+      },
+    });
+
+  }
+
+  componentWillUnmount(){
+    this.gameUpdateSubscription.unsubscribe()
+    this.gameCreationSubscription.unsubscribe()
+    //this.gameDeletionSubscription.unsubscribe()
+  }
+
+  createGame = async (event, gameInfo) => {
+    this.setState({showDialog: false})
+    gameInfo['fen'] = "init"
+    gameInfo['available'] = true
+    let userInfo = await Auth.currentUserInfo()
+    if(userInfo) {
+      let user = {}
+      user.id = userInfo.id
+      user.username = userInfo.username
+      gameInfo['creator'] = user
     }
+    let currentGame = localStorage.getItem(CURRENT_GAME)
+    if(currentGame){
+       API.graphql(graphqlOperation(mutations.deleteGame, { input: {id: currentGame }}))
```

Comment:
What if a user wants to play multiple games at the same time?

Reply:
The first design actually allowed users to create multiple games. But i figured that if a user creates a new game, it's safe to assume that they're no longer interested in the old one. Furthermore, it could lead to complications and make the code unnecessarily complicated.

Reply:
This is a group project, you can’t just implement everything the way you want without talking to your teammates. In an agile team what a product looks like and how it should function is up to the product owner, not to the developer.